### ISSUE TITLE
Sort recipes by creation date or update date (INC/DEC)

### ISSUE THREAD

**Issue by kelson42** [openzim/zimfarm#675]
Following https://github.com/openzim/cms/issues/3#issuecomment-766685594, it seems we need a way to sort recipes. At least to get the latest created recipes.

**Comment by Mightymanh** [comment]
I can attempt this issue. I think it is simple to sort recipes based on dates. 
1. How do I get assigned to this issue? 

2. Based on the CONTRIBUTING.md in the root folder, and based on this task, I think I need to set up in my local machine the following parts to be able to work on this issue:
- backend API
- frontend UI

Is there any other parts that you think I may miss?

**Comment by benoit74** [comment]
Issue assigned, welcome.

You will need a DB as well.

The simplest solution is to use the docker-compose stack in `dev` folder (see README.md in this folder for details).

**Comment by Mightymanh** [comment]
I followed the instructions in the README.md in dev.

![Image](https://github.com/user-attachments/assets/ac8d8381-0e63-455d-a241-d344e9c68c02)

try to run the second command but get the error below:
<img width="857" alt="Image" src="https://github.com/user-attachments/assets/82663f1f-7677-466f-81a3-2566240d5f2b" />
Looks like the front-end docker image does not exist. How should I resolve this? I'm new to Docker so right now what I'm doing is learning Docker and try to understand the Docker script.

**Comment by benoit74** [comment]
Please sort them by date by default

**Comment by Mightymanh** [comment]
I get the following containers to run: 

![Image](https://github.com/user-attachments/assets/33504554-2e0b-4496-96cc-9aff3239835c)

I get the database up to date and populate with some data according to the ipynb instruction file in the dev folder.

However, the backend container for some reason does not want to run after many tries. I check the log and it seems like there is a command in the prestart.sh that fails the backend container: 

![Image](https://github.com/user-attachments/assets/d42ac431-8f06-473c-b24a-6c50b8c68c2d)

I check the prestart.sh file that links to the prestarts.sh in the container. There is no such $'\r' command.

![Image](https://github.com/user-attachments/assets/2542d36e-c038-48b5-be72-025cb5e77733)

Any idea on how to resolve this issue? I'm new to Docker and I'm trying my best to understand what is going on. I think the prestart.sh in the github directory is different than the prestart.sh in the container directory

**Comment by benoit74** [comment]
Not much to share unfortunately ... looks like a problem of carriage return, probably a bad cooperation between Windows and Linux ... can't repro on my side

**Comment by Mightymanh** [comment]
![Image](https://github.com/user-attachments/assets/c94cb0b0-7208-4d70-91da-bffea640f715)

Is the creation date of a recipe in the duration shown in highlight?

**Comment by benoit74** [comment]
Indeed, we do not store creation nor update date of recipes in the DB. And this is not the correct date (only an good approximation). 

And I'm now sure this issue is in fact not ready to be developed at all:
- currently UI does not allow to sort recipes on custom criteria, and I'm pretty sure that sometime we will want to sort them by name, and sometime by creation date, and sometime by last update date => this means lots of changes in UI and in API ; is that what we want ?
- do we want to add creation date and update date in DB (and API ? and UI ?) ?
- which event should be considered as an update ? any change to recipe configuration ? including simple changes like image version (and I'm pretty sure there are other "benign" changes we might not want to consider as an update of the recipe) ?

@Mightymanh I'm sorry but this is probably not that a good first issue at all

**Comment by Mightymanh** [comment]
Glad to hear about the confirmation. I think that adding creation date in DB seems not bad. The problem is how do we add a new recipe? When we add a new recipe, we can always track the current time, convert to standard UTC timestamp, and store in database as "creation date". 

Btw, I notice that the UI does not implement proper pagination. In the image below it shows the first 10 results out of 2601 results. But there is no way to move on to the next 10 results. Maybe something I can help improving the UI?

![Image](https://github.com/user-attachments/assets/acbfb26d-96c7-4d0a-b798-8e19fdd59b51)

**Comment by Mightymanh** [comment]
@benoit74 any update to previous comment? I'm trying to score a pull request for gsoc program so if this any of above idea sounds not good then I can move on to other issues.

**Comment by benoit74** [comment]
You're better moving on to another issue, as I said this issue is probably not ready to be developed, we still have open questions I don't know how to answer

**Comment by benoit74** [comment]
@Mightymanh for instance https://github.com/openzim/zimfarm/issues/898 is both very easy, very clearly defined, a top prio and a regression.
